examples/plot_openfmri_ds001.py

```diff
@@ -0,0 +1,214 @@
+""" Example of a simple GLM fit on an OpenfMRI dataset subject
+
+The script defines a General Linear Model and fits it to OpenfMRI data.
+
+Author: Martin Perez-Guevara, 2016
+"""
+
+import os
+import glob
+import itertools
+import numpy as np
+import pandas as pd
+import nibabel as nib
+from nilearn.plotting import plot_stat_map, show
+from sklearn.datasets.base import Bunch
+
+from nistats.design_matrix import make_design_matrix
+from nistats.glm import FirstLevelGLM
+from nistats.datasets import fetch_openfmri_dataset
+
+
+# Utility to load data from openfmri dataset for a subject
+
+# openfmri dataset extraction
+OPENFMRI_SEP = '\t'
+OTHER_FILE_SEP = ' '
+
+
+def _glob_openfmri_data(dataset_dir, sub_id, model_id, task_id):
+    '''Extracts model from openfmri dataset for given task and subject'''
+    _subject_data = {}
+    onsets_template = os.path.join(dataset_dir,
+                                   'sub{0:03d}'.format(sub_id),
+                                   'model',
+                                   'model{0:03d}'.format(model_id),
+                                   'onsets',
+                                   'task{0:03d}'.format(task_id) + '_run{0}',
+                                   '{1}.txt')
+    condkey_file = os.path.join(dataset_dir, 'models',
+                                'model{0:03d}'.format(model_id),
+                                'condition_key.txt')
+    cond_df = pd.read_csv(condkey_file,
+                          sep=OTHER_FILE_SEP,
+                          header=None)
+    info_conditions = cond_df[cond_df[0] == 'task{0:03d}'.format(task_id)]
+    conds = info_conditions[1].tolist()
+    conds_name = info_conditions[2].tolist()
+    run_path = os.path.join(dataset_dir, 'sub{0:03d}'.format(sub_id),
+                            'BOLD', 'task{0:03d}_run*'.format(task_id),
+                            'bold.nii.gz')
+    runs = glob.glob(run_path)
+    runs.sort()
+    TR = float(open(os.path.join(dataset_dir, 'scan_key.txt'))
+               .read().split(OTHER_FILE_SEP)[1])
+    _run_events = {}
+    all_event_files = []
+
+    for run in runs:
+        names = []
+        allonsets = []
+        alldurations = []
+        event_files = []
+        # This will be used to add dummy events.
+        vol = nib.load(os.path.join(dataset_dir, 'sub{0:03d}', 'BOLD',
```

Comment:
vol is  a bad name for the number of TRs.
Rather use Nilearn than nibabel when loading data.